BuffTrigger.lua

```diff
@@ -296,70 +295,84 @@ do
 end
 WeakAuras.aura_cache = aura_cache;
 
-function WeakAuras.SetAuraVisibility(id, triggernum, data, active, unit, duration, expirationTime, name, icon, count, cloneId, index, spellId)
-  local region;
-  local showClones;
+function WeakAuras.SetAuraVisibility(id, triggernum, cloneId, inverse, active, unit, duration, expirationTime, name, icon, count, index, spellId)
+  local triggerState = WeakAuras.GetTriggerStateForTrigger(id, triggernum);
 
-  if(cloneId) then
-    if(data.numAdditionalTriggers > 0) then
-      showClones = data.region.toShow and true or false;
-    else
-      showClones = true;
-    end
-    region = WeakAuras.EnsureClone(id, cloneId);
-  else
-    region = data.region;
+  local show = false
+  if(active ~= nil) then
+    if not(inverse and UnitExists(unit)) then
+     show = true;
+   end
+  elseif(inverse and UnitExists(unit)) then
+    show = true;
   end
 
-  region.index = index;
-  region.spellId = spellId;
+  cloneId = cloneId or "";
 
-  local show;
-  if(active ~= nil) then
-    if not(data.inverse and UnitExists(unit)) then
-      show = true;
-    end
-  elseif(data.inverse and UnitExists(unit)) then
-    show = true;
+  if (not show and not triggerState[cloneId]) then
+    return false;
   end
 
-  if(show) then
-    if(triggernum == 0) then
-      if(region.SetDurationInfo) then
-        region:SetDurationInfo(duration, expirationTime > 0 and expirationTime or math.huge);
-      end
+  triggerState[cloneId] = triggerState[cloneId] or {};
```

Comment:
This triggerState here is the basic structure in which all the dynamic information of each trigger is stored, this code here instead of directly modifying the region, stores all the information in that state and finally calls UpdatedTriggerState, which applies it to the region.